### PR TITLE
Java docs Part 1

### PR DESCRIPTION
**Pull request by conniey**
* Removing unneeded public API
  * enum LockRenewalStatus
  * CompleteOptions.self() method
* Updating javadocs for: 
    * `*Options` classes
    * Administration classes
    * First set of main classes (Connection properties and builder.)
* Updating codesnippets for processors. Removing redundant/overlapping code.

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java [hunk 1]
```diff
@@ -639,17 +640,23 @@ public ServiceBusSenderClient buildClient() {
...
+     * By default, the processor:
+     * <ul>
+     *     <li>Automatically settles messages. Disabled via {@link #disableAutoComplete()}</li>
+     *     <li>Processes 1 session concurrently. Configured via {@link #maxConcurrentSessions(int)}</li>
+     *     <li>Invokes 1 instance of {@link #processMessage(Consumer) processMessage consumer}. Configured via
+     *     {@link #maxConcurrentCalls(int)}</li>
+     * </ul>
+     * {@link #maxConcurrentCalls(int)} (int) sessions concurrently}. By default, the
      * processor will receive messages from a single session with rollover.
+     * </p>
```
Comment (JonathanGiles) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java:656]: Is there something weird here?

Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java:656]: I thought I caught that one. thanks!

**Review by srnagar — APPROVED** [review]
LGTM!

**Review by hemanttanwar — APPROVED** [review]

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java [hunk 2]
```diff
@@ -639,17 +640,21 @@ public ServiceBusSenderClient buildClient() {
...
+     * entity. {@link ServiceBusProcessorClient} processes messages and errors via {@link #processMessage(Consumer)}
+     * and {@link #processError(Consumer)}. When the processor finishes processing a session, it tries to fetch the
+     * next session to process.
+     *
+     * <p>
+     * By default, the processor:
+     * <ul>
+     *     <li>Automatically settles messages. Disabled via {@link #disableAutoComplete()}</li>
+     *     <li>Processes 1 session concurrently. Configured via {@link #maxConcurrentSessions(int)}</li>
+     *     <li>Invokes 1 instance of {@link #processMessage(Consumer) processMessage consumer}. Configured via
```
Comment (hemanttanwar) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java:652]: Should we explain about lock renewal also here ?

Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java:652]: We don't have that ability in the session processor? I don't think we have auto renew session lock.

**Pull request merged**